lib/src/modules/token_storage.dart

```diff
@@ -87,13 +87,35 @@ class TokenStorage {
       [String? resource, List<String>? scopes]) async {
     final key = _buildAccessTokenKey(resource, scopes);
 
-    if (_accessTokenMap != null) {
-      return _accessTokenMap![key];
+    _accessTokenMap ??= await _getAccessTokenMapFromStorage();
+
+    final accessToken = _accessTokenMap?[key];
+
+    // remove the access token if expired and return null
+    if (accessToken?.isExpired == true) {
+      await _deleteAccessToken(key);
+      return null;
     }
 
-    _accessTokenMap = await _getAccessTokenMapFromStorage();
+    return accessToken;
+  }
+
+  Future<void> _deleteAccessToken(String accessTokenKey) async {
+    final tempAccessTokenMap = _accessTokenMap;
```

Comment:
This kind of temp variable does not work in dart. Both variables will still refer to the same address in memory. To solve that you should create a new map by either calling "Map.from(_accessTokenMap)" or "{..._accessTokenMap}".

Reply:
updated Thx